generator/templates/client/parameter.gotmpl

```diff
@@ -65,6 +65,19 @@ func New{{ pascalize .Name }}ParamsWithContext(ctx context.Context) *{{ pascaliz
   }
 }
 
+// New{{ pascalize .Name }}ParamsWithHTTPClient creates a new {{ pascalize .Name }}Params object
+// with the default values initialized, and the ability to set a custom HTTPClient for a request
+func New{{ pascalize .Name }}ParamsWithHTTPClient(client *http.Client) *{{ pascalize .Name }}Params {
```

Comment:
@casualjim isn't that why you have the WithContext function, for situations like that? Otherwise one would have to provide N constructors, where N is the amount of combinations possible.